test/test_shellfileprocessor.py

```diff
@@ -0,0 +1,21 @@
+
+import os
+from fetch._core import ShellFileProcessor
+
+
+def test_shellfilepro_required_files_there():
+    command = 'ls {base}.py'
+    # required_files = (r'^(?P<base>.*test.+)\.py$',['{base}.py','{base}.py'])
+    required_files = ('^(?P<base>.*test.+)\\.py$', ['{base}.py', '{base}.py'])
+    file_path = os.path.abspath(__file__)
+    sfp = ShellFileProcessor(command=command, expect_file=file_path, required_files=required_files)
+    results = sfp.process(file_path)
+
+def test_shellfilepro_required_files_not_there():
+    command = 'ls {base}.py'
+    required_files = (r'^(?P<base>.*test.+)\.py$', ['{base}.py', '/this/is/not/here/please.py'])
+    file_path = os.path.abspath(__file__)
+    sfp = ShellFileProcessor(command=command, expect_file=file_path, required_files=required_files)
+    results = sfp.process(file_path)
+    # A rather toothless assert given required_files_there would return the same result...
+    assert file_path == results
```

Comment:
One option is to ask for a [tmp_path](https://docs.pytest.org/en/latest/tmpdir.html), and make your command `touch {tmp_path / 'test_file.txt'}`. Then if the file exists, you know the command was run. It could work in both tests.

I like your instinct to add a test for both "there" and "not there".  Maybe we also need one for when no `required_files` parameter is supplied at all? (where it should always run the command).

Reply:
I ran out of time to implement this.  I've added it into the comments.

Reply:
Meaning I've added your idea's into comments in the test py file...